### DISCUSSION TITLE
Identify specific co-mutational patterns in a specific TCGA cancer type through maftools

### DISCUSSION THREAD

**Discussion by Jasonmbg** [PoisonAlien/maftools#735] [Q&A]
Dear Anand,

I would like to ask a particular question concerning the robust utilization of the R packages maftools and TCGAmutations:

in particular, based on a current project analysis, we are aiming on identify putative co-mutation/occurrence patterns based on TP53 mutations in breast cancer and other putative cancer drivers, in order to prioritize possible co-dependencies which could sensitize breast cancer cell lines and validate putative experimental designs; on this premise, based on the available TCGA breast cancer cohort from TCGAmutations, my main methodological questions are the following:

1) In order to specifically identify significant "TP53-pair" interactions in the maf object cohort, would in your opinion simply  the `somaticInteractions` function as illustrated here would suffice? 
http://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html#9_Analysis 

And then specifically check any significant p-values that include as one gene the TP53? and as event "co-occurrence"?

2) If some initial filters would be applied prior running the above function, for example excluding any genes with **overall mutational frequency of less than 0.1**, is this possible through maftools? In addition, to include only specific TP53 hotspot mutations, such as **R273H** ?

3) A parallel analysis based on your expertise, could be to separate the maf object into two distinct "maf" populations; that is TP53 mutated, and wild type TP53? and identify the top differentially mutated genes, and investigate also any mutational patterns based on oncoplots?

Thank you in advance,

Efstathios

**Comment by PoisonAlien** [comment]
Hello Efstathios,

1. Yes, `somaticInteractions` should be enough IMO.
2. You could try `subsetMaf` for such operations.
3. Sure, you could do this as well. It's a valid approach..

**Reply by Jasonmbg** [reply]
Dear @PoisonAlien,

thank you very much for your confirmation and directions-however, I have to ask if possible for some further directions, especially concerning steps 1 and 2:

1) In detail, regarding the subsetting of maf objects:

A) Initially, except keeping specific genes, if I would like also to use the field query, to additionally filter for specific protein coding mutations based on the HGVSp_short:

```
brca.tp53 <- subsetMaf(brca.maf,genes="TP53", query= "HGVSp_Short%in%c('p.R273H',p.R2495')")
Error in parse(text = query) : <text>:1:35: unexpected INCOMPLETE_STRING
1: HGVSp_Short%in%c('p.R273H',p.R2495')
                                      ^
```

Thus, how the above could be fixed to use multiple fields for a specific maf column?

B) For my second part as it is more complicated-in order to utilize `somaticInteractions`, I would like to use T53-mutated samples-but with specific mutations as above-along with wild type samples-in your expertise, the only possible way is to subset initially the maf object into two separate maf objects, and then merge them by `merge_mafs` ? 

Best,

Efstathios

**Reply by ShixiangWang** [reply]
You missed a `'` before `p.R2495'`

**Reply by Jasonmbg** [reply]
Dear @ShixiangWang,

thank you very much for pointing this out :)

**Comment by Jasonmbg** [comment]
Dear @PoisonAlien,

unfortunately based on the subsetting of MAF objects I'm facing another error-in detail when I'm trying to subset the following MAF object into two separate MAF objects and concatenate them at a later step:

```r
TCGAmutations::tcga_available()
brca.maf <- TCGAmutations::tcga_load(study = "BRCA")

# possible approach to filter genes with overall low mutational frequency

dt <- brca.maf@data

dat.sel <- dt %>% distinct(Hugo_Symbol,Tumor_Sample_Barcode) %>% mutate(value=1) %>% pivot_wider(names_from = "Tumor_Sample_Barcode", values_from="value",values_fill=0)  

to.plot <- data.table(mutation = dat.sel$Hugo_Symbol,
frequency = rowMeans(dat.sel[,-1],na.rm=T)) # perhaps filter based on the total frequency-for example, < than 2%


pass.freq <- to.plot[to.plot$frequency>=0.01,]
pass.freq.genes <- pass.freq$mutation


maf.brca.selgenes <- subsetMaf(brca.maf, genes = pass.freq.genes, mafObj = T, isTCGA = T)

brca.tp53.sel <- subsetMaf(maf.brca.selgenes,genes="TP53", query= "HGVSp_Short%in%c('p.R273H','p,R248W', 'p.R273C', 'p.R273L','p.R248Q','p.R175H')", mafObj = T, isTCGA = T)

all_samples = levels(getSampleSummary(x = maf.brca.selgenes)[,Tumor_Sample_Barcode])

TP53.hotspots <- levels(getSampleSummary(x = brca.tp53.sel)[,Tumor_Sample_Barcode])

TP53.mutants = genesToBarcodes(maf =maf.brca.selgenes, genes = "TP53", justNames = TRUE)

dt <- setdiff(TP53.mutants$TP53,TP53.hotspots)

dt2 <- setdiff(all_samples,dt)

braf_maf_other = subsetMaf(maf = maf.brca.selgenes, tsb = dt2, mafObj = T, isTCGA = T)

Error in subsetMaf(maf = maf.brca.selgenes, tsb = dt2, mafObj = T, isTCGA = T) : 
  Subsetting has resulted in zero non-synonymous variants!

sessionInfo()
R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] forcats_0.5.1       stringr_1.4.0       dplyr_1.0.6         purrr_0.3.4        
 [5] readr_1.4.0         tidyr_1.1.3         tibble_3.1.2        ggplot2_3.3.4      
 [9] tidyverse_1.3.1     TCGAmutations_0.3.0 data.table_1.14.0   maftools_2.6.05 

```

Thus, why this is returning an error? My ultimate goal was to combine the maf samples that had specific TP53 hotspot mutations, along with the rest of the samples that do not have any TP53 mutation, in order then to search for putative somatic interactions-however, I can't identify where the error occurs in which of the subset steps;

Thank you in advance,

Efstathios

**Reply by PoisonAlien** [reply]
Hi,

In the below command, don't use is `TCGA = TRUE` and it works fine.

```r
braf_maf_other = subsetMaf(maf = maf.brca.selgenes, tsb = dt2, mafObj = T, isTCGA = T)
```

I am swamped with work, please be patient as it takes a while for me to answer.

**Reply by Jasonmbg** [reply]
Dear @PoisonAlien  ,

first of all thank you very much for your help, and I really appreciate your time spent, I fully understand you !! :)

Something last to verify, in order not to disturb you further and be certain of the correct implementation:

From the above step, aiming to perform the final analysis of somatic interactions, including the selected TP53 variants and the rest of the mutated genes, the most "safe" approach would be:


```
braf_maf_other = subsetMaf(maf = maf.brca.selgenes, tsb = dt2, mafObj = T) # without TCGA as you mentioned
maf.list <- list(brca.tp53.sel,braf_maf_other) # where brca.TP53.sel from above

final_merged_maf <- merge_mafs(maf.list)
Merging 2 MAF objects
-Validating
--Removed 775 duplicated variants
-Silent variants: 5191 
-Summarizing
--Possible FLAGS among top ten genes:
  TTN
  MUC16
  FLG
-Processing clinical data
-Finished in 0.670s elapsed (0.660s cpu) 

```
Kind Regards,

Efstathios

**Reply by PoisonAlien** [reply]
No need to be sorry, happy to help :)
Yes, the merging step sounds good to me.
